### ISSUE TITLE
bigint throws an error during sails lift

### ISSUE THREAD

**Issue by jamesflorentino** [balderdashy/sails-postgresql#167]
Hi, I'm currently getting an error whenever I set a model attribute to `bigint`. Here's what happens when I do a `sails lift`.

```
Error (E_VALIDATION) :: 1 attribute is invalid
    at WLValidationError.WLError (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/error/WLError.js:26:15)
    at new WLValidationError (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/error/WLValidationError.js:20:28)
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/query/validate.js:45:43
    at allValidationsChecked (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/core/validations.js:198:5)
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:49:16
    at done (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:239:19)
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:40:16
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/core/validations.js:179:23
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:49:16
    at done (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:239:19)
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:40:16
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/lib/waterline/core/validations.js:152:64
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:162:20
    at /Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:230:13
    at _arrayEach (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:81:9)
    at _each (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:72:13)
    at Object.async.forEachOf.async.eachOf (/Users/james/.nvm/versions/node/v0.12.1/lib/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:229:9)

Invalid attributes sent to Bet:
 • amount
   • [object Object]
```

Here's what my model looks like:

```
  attributes: {
    amount: 'bigint',

    selection: {
      model: 'selection',
      required: true
    },

    address: {
      type: 'string',
      required: true
    },

    match: {
      model: 'match',
      required: true
    }
  }
```

Can anyone shed some light? My postgres version is 0.9.4.4. `    "sails-postgresql": "^0.10.15"`,`    "sails": "~0.11.0"`.

Appreciate your help. Thanks!

**Comment by devinivy** [comment]
This probably means anchor, waterline's validation library, doesn't like seeing "bigint," which is not a standard waterline type.  sails-mysql deals with this problem by setting `type` to `integer` then paying attention to the attribute's `size` property to decide on using BIGINT, INT, or TINYINT.

For starters, inspect that error further to see what the validation error is (where it says `[object Object]` there's actually an object there with more information).  Here's the standard set of waterline types: https://github.com/balderdashy/waterline-docs/blob/master/models/data-types-attributes.md

There's also a PR in the works to support custom database types in a more unified way.

**Comment by jamesflorentino** [comment]
@devinivy Thanks for the reply. I'll give it a shot. :+1:

**Comment by sailsbot** [comment]
Thanks for posting, @jamesflorentino.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
